src/ch/ch_monitor.c

```diff
@@ -913,18 +917,40 @@ static int virCHMonitorReadProcessEvents(virCHMonitorPtr mon)
     return (events > 0) ? virCHMonitorProcessEvents(mon, events) : events;
 }
 
+#define MONITOR_TMP_FAIL_RETRIES 5
+
 static void virCHMonitorEventLoop(void *data)
 {
     virCHMonitorPtr mon = (virCHMonitorPtr)data;
-    virDomainObjPtr vm = mon->vm;
+    virDomainObjPtr vm = NULL;
+    int retries = 0;
+    int monitor_fd;
 
     VIR_DEBUG("Monitor event loop thread starting");
 
+    while((monitor_fd = open(mon->monitorpath, O_RDONLY)) < 0) {
+        if ((errno == ENOMEM && ++retries < MONITOR_TMP_FAIL_RETRIES) ||
```

Comment:
I'm not sure if this should timeout fail or we should just wait but I'm fine keeping it this way for now.

Reply:
Initially I coded it like that(wait on ENOMEM as well). But then I felt that, if linux cannot reclaim memory soon, it might need some manual intervention and its better bail out. I can probably increase the wait time to 1 second per retry so that we wait 5 seconds before bailing out.

Reply:
Increased the wait time to 1 sec per retry.

Reply:
That `ENOMEM` would only really be for the per-user limit on pipe memory allocation, if we hit that once I can see the argument for just quitting with a descriptive error message. Linux itself will happily over commit on memory (and then the OOM killer sorts things out anyway if there is an issue) so we shouldn't see an error due to general memory allocation.

Reply:
Ahh yes, ENOMEM for open happens only if we cross the per-usre limit. I shall remove this retry logic then